File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/ViewpagerActivity.java
package ar.com.wolox.android.example.ui.viewpager;

import ar.com.wolox.android.R;
import ar.com.wolox.android.example.ui.viewpager.fragment.ViewPagerFragment;
import ar.com.wolox.wolmo.core.activity.WolmoActivity;

import javax.inject.Inject;

import dagger.Lazy;

public class ViewpagerActivity extends WolmoActivity {

    // Lazy example, a lazy injection does not build the dependencies until #get() is called
    @Inject Lazy<ViewPagerFragment> mLazyViewpagerFragment;

    @Override
    protected int layout() {
        return R.layout.activity_base;
    }

    @Override
    protected void init() {
        replaceFragment(R.id.activity_base_content, mLazyViewpagerFragment.get());
    }
}

File: app/src/main/java/ar/com/wolox/android/example/ui/viewpager/ViewpagerActivity.kt
package ar.com.wolox.android.example.ui.viewpager

import ar.com.wolox.android.R
import ar.com.wolox.android.example.ui.viewpager.fragment.ViewPagerFragment
import ar.com.wolox.wolmo.core.activity.WolmoActivity
import dagger.Lazy
import javax.inject.Inject

class ViewpagerActivity @Inject constructor() : WolmoActivity() {

    // Lazy example, a lazy injection does not build the dependencies until #get() is called
    @Inject internal lateinit var lazyViewpagerFragment: Lazy<ViewPagerFragment>

    override fun layout(): Int = R.layout.activity_base

    override fun init() {
        replaceFragment(R.id.vActivityBaseContent, lazyViewpagerFragment.get())
    }
}
